src/plugins/data/public/actions/filters/create_filters_from_value_click.ts

```diff
@@ -148,3 +156,40 @@ export const createFiltersFromValueClickAction = async ({
     compareFilters(a, b, COMPARE_ALL_OPTIONS)
   );
 };
+
+/** @public */
+export const appendFilterToESQLQueryFromValueClickAction = async ({
+  data,
+  query,
+}: ValueClickDataContext) => {
+  let queryWithFilter: string | undefined;
+
+  await Promise.all(
+    data
+      .filter((point) => point)
+      .map(async (val) => {
+        const { table, column: columnIndex, row: rowIndex } = val;
+        if (table && table.columns && table.columns[columnIndex]) {
+          const column = table.columns[columnIndex];
+          const value: unknown = rowIndex > -1 ? table.rows[rowIndex][column.id] : null;
+
+          // Do not append in case of time series, for now. We need to find a way to compute the interval
+          // to create the time range filter correctly. The users can brush to update the time filter instead.
+          const isDate = column.meta?.type === 'date';
+
+          if (value == null || isDate || !query) return;
+
+          const queryString = query.esql;
+          queryWithFilter = appendWhereClauseToESQLQuery(
+            queryString,
+            column.id,
```

Comment:
I'm running into a scenario where `column.id` is a GUID, not a column name.... And when I click to filter I get an error:

https://github.com/elastic/kibana/assets/1178348/c97e153f-5511-444e-ac51-ecc4617ea54a

Is it possible this should be `column.name`? I'm trying to come up with reproducible instructions... I'll post them here when I have them.

Reply:
Okay I found out how to reproduce it:

1. Install sample web logs
2. Go to Discover & switch to ES|QL mode
3. Change the date range to something with data (I used "Today")
4. Enter the following query:

```
from kibana_sample_data_logstsdb
| stats total_memory = sum(memory) by request
```

5. In the Lens config, change the "Breakdown" field to be the same as the "Horizontal axis" (should be "request")
6. Click "Apply & close"
7. Click on one of the bars

Here's the error:

```
[esql] > Unexpected error from Elasticsearch: verification_exception - Found 1 problem
line 3:9: Unknown column [c8a7bd5e-8712-4e2c-90b2-f87be86603b3]
```

I think it has to do with the fact that the same column is used multiple times in the Lens config.

Reply:
Fantastic catch @lukasolson and thanx for the replication steps, I was using the id instead of the column name. Fixed https://github.com/elastic/kibana/commit/ec359541da318457a1509bf19732a1269cc4ac4c